docs/cloud/azure/architecture.md

```diff
@@ -1 +1,937 @@
+operates as a link between the client’s data feed and a number of
+subscribers. It implements a pub-sub pattern, specifically, it receives
+data from the feedhandler, stores it locally in a table then saves it
+to a log file. It publishes this data to a realtime database (RDB) and
+any clients who have subscribed to it. It then purges its local tables
+of data.
+
+Tickerplants can operate in two modes:
+
+mode | operation
+-----|----------
+batch | collects updates in its local tables; batches up for a period of time and then forwards the update to realtime subscribers in a bulk update
+realtime (zero&nbsp;latency) | forwards the input immediately: requires smaller local tables but has higher CPU and network costs – each message has a fixed network overhead
+
+API calls:
+
+call | operation
+-----|----------
+subscribe | add subscriber to message receipt list and send subscriber table definitions
+unsubscribe | remove subscriber from message receipt list
+
+End of Day event: at midnight, the TP closes its log files, auto creates a new file, and notifies the realtime database about the start of the new day.
+    
+
+### Realtime database
+
+The realtime database holds all the intraday data in memory, to
+allow for fast powerful queries. For example, at the start of business
+day, the RDB sends a message to the tickerplant and receives a reply
+containing the data schema, the location of the log file, and the number
+of lines to read from the log file. It then receives subsequent updates
+from the tickerplant as they are published. One of the key design
+choices for <!-- Amazon Web Services --> Microsoft Azure is the size of memory for this instance,
+as ideally we need to contain the entire business day/period of data
+in-memory.
+
+Purpose:
+
+-   Subscribed to the messages from the tickerplant
+-   Stores (in-memory) the messages received
+-   Allows this data to be queried intraday
+
+Actions:
+
+-   On message receipt: insert into local, in-memory tables.
+-   End of Day receipt: usually writes intraday data down then sends a
+    new End-of-Day message to the HDB. Optionally RDB sorts certain
+    tables (e.g. by sym and time) to speed up queries.
+
+An RDB can operate in single or multi-input mode. The default mode is
+single input in which user queries are served sequentially and queries
+are queued till an update from the TP is processed (inserted into the
+local table).
+
+In standard tick scripts, the RDB tables are indexed, typically by the
+product identifier. An index is a hash table behind the scene. Indexing
+has a significant impact on the speed of the queries at the cost of a
+bit slower ingestion. The insert function takes care of the indexing,
+i.e. during an update it also updates the hash table.
+
+Performance of the CPU and memory in the chosen Azure VM instance impacts on the overall sustainable rates of ingest and
+queryable rate of this realtime kdb+ function.
```

Comment:
Doesn't quite make sense to me. Plus, "queryable" is not a word. How does this sound:

```suggestion
The performance of the CPU and memory in the chosen Azure VM instance impacts the rate at which data is ingested and the time taken to execute data queries.
```

Reply:
Sounds good to me.

On a side note, I guess the language in IT is quite alive https://en.wiktionary.org/wiki/queryable